eng/scripts/CodeCheck.ps1

```diff
@@ -79,6 +79,22 @@ try {
         }
     }
 
+    #
+    # Check for unexpected (not from dotnet-public-npm) yarn resolutions in lock files.
+    #
+
+    $registry = 'https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public-npm/npm/registry/'
+    Get-ChildItem src\yarn.lock -Recurse |
+    ForEach-Object FullName |
+    Where-Object {$_ -NotLike '*\node_modules\*'} |
+    Where-Object {
+        Select-String '^  resolved ' $_ | Select-String -quiet -NotMatch $registry
+    } |
+    ForEach-Object {
+        LogError -filepath $_ `
+            "Packages in yarn.lock file resolved from wrong registry. All dependencies must be resolved from $registry"
```

Comment:
Can you print the text of the line that caused the error here? Many devs aren't familiar w/ npm stuff & that'll make it easier for them to find the culprit change

Reply:
Done in new last commit.